### ISSUE TITLE
Makefile target 'make doc' is broken 

### ISSUE THREAD

**Issue by return42** [searx/searx#2278]
PR #2256 breaks the docs build

````
$ make docs
build searx/brand.py
build utils/brand.env
PYENV     usage: source ./local/py3/bin/activate
PYENV     using virtualenv from ./local/py3
PYENV     install .
SPHINX    docs --> file:///800GBPCIex4/share/searx/dist/docs
Running Sphinx v3.2.1
ERROR:searx:server.secret_key is not changed. Please use something else instead of ultrasecretkey.

Configuration error:
The configuration file (or one of the modules it imports) called sys.exit()
make: *** [Makefile:84: docs] Error 2
````

FWIW: a workaround is to use debug mode:

      SEARX_DEBUG=1 make docs-clean docs-live

**Comment by return42** [comment]
Same with `make run` / reloading webapp.py when a file is changed exit with error

````
$ make run
....
INFO:werkzeug: * Detected change in '/800GBPCIex4/share/searx/searx/engines/deviantart.py', reloading
INFO:werkzeug: * Restarting with stat
ERROR:searx:server.secret_key is not changed. Please use something else instead of ultrasecretkey.
make: *** [Makefile:69: run] Error 1
````

Further read : https://github.com/searx/searx/issues/2252#issuecomment-706680323

**Comment by dalf** [comment]
If these line 
https://github.com/searx/searx/blob/54afe37e18cdbf4fd858907f0a6e83c6cb1c4157/searx/__init__.py#L89-L91

are moved just after 
https://github.com/searx/searx/blob/54afe37e18cdbf4fd858907f0a6e83c6cb1c4157/searx/webapp.py#L1042

would it be still okay ?

[EDIT] `make run` should be updated to use `SEARX_DEBUG=1 $(PY_ENV)/bin/python ./searx/webapp.py` instead of changing the default settings.yml (and now I understand why the reload feature was working only once: in the second load of searx, debug is False in settings.yml)

----

otherwise, as an alternative, I suggest to make mandatory this plug-in : https://github.com/asciimoo/searx_external_plugin_example/blob/master/plugin1/resources/bg.js (without a blinking background, but a big warning on every HTML pages using CSS)

**Comment by return42** [comment]
> make run should be updated to use SEARX_DEBUG=1 

Good point! I prepared PR #2287 / do you have time to review? / thanks!

**Comment by return42** [comment]
closed by  #2411
